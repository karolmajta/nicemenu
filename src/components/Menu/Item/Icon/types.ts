export type IconName =
  | "bolt"
  | "dot"
  | "inbox"
  | "loader"
  | "pencil"
  | "reschedule";

export interface IconProps {
  name: IconName;
  className?: string;
  classNames?: Array<string>;
}
