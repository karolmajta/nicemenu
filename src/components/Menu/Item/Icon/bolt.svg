<svg width="15" height="16" viewBox="0 0 15 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<g opacity="0.5" clip-path="url(#clip0)">
<path d="M8.66667 2.33337L2 10.3334H8L7.33333 15.6667L14 7.6667H8L8.66667 2.33337Z" stroke="#E4E4EE" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
</g>
<defs>
<clipPath id="clip0">
<rect width="15" height="16" fill="white"/>
</clipPath>
</defs>
</svg>
