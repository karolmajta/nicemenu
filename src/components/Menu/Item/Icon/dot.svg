<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<g opacity="0.5" clip-path="url(#clip0)">
<circle cx="8" cy="8" r="8" fill="#0E17B5"/>
<ellipse cx="7.99999" cy="7.99999" rx="6.46957" ry="6.46956" fill="url(#paint0_linear)"/>
<circle cx="7.99996" cy="7.99998" r="4.5913" fill="#EDEFF9"/>
</g>
<defs>
<linearGradient id="paint0_linear" x1="7.99999" y1="1.53043" x2="7.99999" y2="14.4696" gradientUnits="userSpaceOnUse">
<stop stop-color="#1146EB"/>
<stop offset="1" stop-color="#0E17B5"/>
</linearGradient>
<clipPath id="clip0">
<rect width="15" height="16" fill="white" transform="translate(0.5)"/>
</clipPath>
</defs>
</svg>
