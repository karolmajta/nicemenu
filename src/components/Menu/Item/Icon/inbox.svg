<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<g opacity="0.5" clip-path="url(#clip0)">
<path d="M15.8935 6.97005L14.1354 3.41844C13.8335 2.79691 13.212 2.40623 12.5017 2.40623H3.48058C2.77026 2.40623 2.16648 2.77915 1.84684 3.41844L0.0887902 6.95229C0.0355161 7.05884 0 7.18315 0 7.2897V12.0666C0 12.9012 0.674806 13.5938 1.52719 13.5938H14.4728C15.3074 13.5938 16 12.919 16 12.0666V7.30746C15.9822 7.18315 15.9467 7.05884 15.8935 6.97005ZM11.081 6.54386C10.7259 6.54386 10.424 6.79247 10.3352 7.14763C10.3174 7.21867 9.90899 9.11878 7.97336 9.11878C6.09101 9.11878 5.66482 7.36073 5.61154 7.14763C5.54051 6.79247 5.23862 6.54386 4.8657 6.54386H1.9889L3.21421 4.09325C3.26748 4.00446 3.37403 3.91567 3.48058 3.91567H12.5017C12.6082 3.91567 12.7148 3.96894 12.768 4.09325L13.9756 6.54386H11.081Z" fill="#E4E4EE"/>
</g>
<defs>
<clipPath id="clip0">
<rect width="15" height="16" fill="white" transform="translate(0.5)"/>
</clipPath>
</defs>
</svg>
