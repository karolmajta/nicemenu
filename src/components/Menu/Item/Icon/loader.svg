<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<g opacity="0.5">
<g clip-path="url(#clip0)">
<path d="M9.71146 0.268549C9.67263 0.206445 9.60996 0.163084 9.53817 0.148589C8.52401 -0.0486574 7.48136 -0.0486574 6.4672 0.148589C6.39541 0.163084 6.33274 0.206445 6.29391 0.268549C6.25367 0.331965 6.24208 0.409377 6.26192 0.481822L6.94704 3.06761C6.98221 3.19926 7.11098 3.28317 7.24559 3.26221C7.74199 3.18489 8.24735 3.18489 8.74375 3.26221H8.78374C8.90766 3.26645 9.01816 3.1847 9.0503 3.06493L9.74342 0.48179C9.76328 0.409408 9.75166 0.331965 9.71146 0.268549Z" fill="#E4E4EE"/>
<path d="M3.26291 8.74841C3.22342 8.50068 3.2038 8.25017 3.20427 7.99932C3.20162 7.74593 3.21946 7.49274 3.2576 7.24223C3.27856 7.10759 3.19462 6.97882 3.06301 6.94367L0.47984 6.25056C0.474248 6.24906 0.468594 6.24772 0.462908 6.24659C0.318519 6.21776 0.178128 6.31145 0.149294 6.45584C-0.0497646 7.46981 -0.0497646 8.51283 0.149294 9.52681C0.175004 9.65292 0.287185 9.74267 0.415861 9.74008C0.438885 9.7427 0.462127 9.7427 0.485182 9.74008L3.06832 9.04697C3.19996 9.01179 3.2839 8.88302 3.26291 8.74841Z" fill="#E4E4EE"/>
<path d="M4.27036 11.0143C3.95615 10.622 3.70683 10.1819 3.53195 9.71075C3.48334 9.58342 3.34648 9.51329 3.21471 9.54815L0.628919 10.2413C0.557755 10.2609 0.4979 10.3091 0.46363 10.3746C0.428642 10.4392 0.421894 10.5154 0.44498 10.5851C0.777463 11.5656 1.29916 12.4714 1.98047 13.2509C2.02914 13.3064 2.09865 13.3392 2.1724 13.3415C2.24297 13.3406 2.31058 13.3129 2.36168 13.2642L4.2544 11.3715C4.35061 11.2742 4.35752 11.1198 4.27036 11.0143Z" fill="#E4E4EE"/>
<path d="M6.28073 12.4805C5.8126 12.3013 5.37612 12.0484 4.98782 11.7314C4.88416 11.6505 4.73702 11.6573 4.64128 11.7474L2.74856 13.6401C2.69664 13.6924 2.66865 13.7638 2.67125 13.8374C2.66347 13.9134 2.68871 13.9891 2.74057 14.0453C3.52015 14.7266 4.42585 15.2483 5.40633 15.5808C5.43464 15.5849 5.46334 15.5849 5.49165 15.5808C5.53541 15.5806 5.57843 15.5696 5.61695 15.5488C5.68237 15.5145 5.7306 15.4547 5.75025 15.3835L6.44336 12.7977C6.47819 12.666 6.40806 12.5291 6.28073 12.4805Z" fill="#E4E4EE"/>
<path d="M9.74354 15.5195L9.05043 12.9363C9.01525 12.8047 8.88649 12.7207 8.75187 12.7417C8.25548 12.819 7.75012 12.819 7.25372 12.7417C7.11908 12.7208 6.99031 12.8047 6.95517 12.9363L6.26205 15.5195C6.243 15.5911 6.25459 15.6674 6.29404 15.7301C6.33287 15.7922 6.39554 15.8355 6.46733 15.85C7.4813 16.0491 8.52433 16.0491 9.5383 15.85C9.61009 15.8355 9.67275 15.7922 9.71158 15.7301C9.75101 15.6674 9.7626 15.5911 9.74354 15.5195Z" fill="#E4E4EE"/>
<path d="M13.2675 13.6428L11.3748 11.7501C11.2782 11.653 11.1238 11.645 11.0176 11.7314C10.6293 12.0484 10.1929 12.3013 9.72471 12.4805C9.59738 12.5291 9.52724 12.666 9.56211 12.7978L10.2339 15.3675C10.2535 15.4387 10.3018 15.4986 10.3672 15.5328C10.4057 15.5536 10.4487 15.5646 10.4925 15.5648C10.5208 15.5689 10.5495 15.5689 10.5778 15.5648C11.5583 15.2323 12.464 14.7106 13.2435 14.0293C13.299 13.9807 13.3318 13.9112 13.3342 13.8374C13.3389 13.7662 13.3149 13.6961 13.2675 13.6428Z" fill="#E4E4EE"/>
<path d="M15.5525 10.3852C15.5182 10.3198 15.4584 10.2716 15.3872 10.2519L12.8014 9.5588C12.6697 9.52393 12.5328 9.59406 12.4842 9.7214C12.305 10.1895 12.0521 10.626 11.7351 11.0143C11.6434 11.1236 11.654 11.2857 11.7591 11.3822L13.6518 13.2749C13.7029 13.3236 13.7705 13.3512 13.8411 13.3522C13.9148 13.3499 13.9844 13.3171 14.033 13.2616C14.7143 12.482 15.236 11.5763 15.5685 10.5958C15.5924 10.5264 15.5866 10.4502 15.5525 10.3852Z" fill="#E4E4EE"/>
<path d="M15.8455 6.47183C15.8444 6.46615 15.8431 6.46049 15.8416 6.4549C15.8034 6.3127 15.6571 6.22842 15.515 6.26659L12.9318 6.9597C12.8002 6.99488 12.7163 7.12364 12.7372 7.25826C12.8154 7.75459 12.8154 8.26007 12.7372 8.75641C12.7163 8.89105 12.8002 9.01982 12.9318 9.05496L15.515 9.74807C15.538 9.7507 15.5613 9.7507 15.5843 9.74807C15.713 9.75066 15.8252 9.66095 15.8509 9.5348C15.9506 9.02906 16.0006 8.5148 16.0002 7.99932C15.9983 7.48633 15.9466 6.97476 15.8455 6.47183Z" fill="#E4E4EE"/>
<path d="M15.5684 5.42419C15.2359 4.44371 14.7142 3.53798 14.0329 2.75843C13.9842 2.70295 13.9147 2.67015 13.841 2.66781C13.7695 2.66115 13.6987 2.6854 13.6464 2.73444L11.7536 4.62715C11.6566 4.72374 11.6485 4.87816 11.735 4.98437C12.0513 5.37021 12.3041 5.80394 12.4841 6.26929C12.5237 6.37181 12.6221 6.43957 12.732 6.43989C12.755 6.44251 12.7783 6.44251 12.8013 6.43989L15.3845 5.74677C15.4556 5.72716 15.5155 5.67889 15.5498 5.61347C15.579 5.55481 15.5857 5.48742 15.5684 5.42419Z" fill="#E4E4EE"/>
<path d="M13.2649 1.96666C12.4853 1.28535 11.5796 0.763651 10.5991 0.431168C10.5294 0.408082 10.4531 0.414861 10.3885 0.449818C10.3231 0.484088 10.2749 0.543912 10.2552 0.615107L9.56211 3.2009C9.52437 3.33064 9.58966 3.46806 9.71406 3.5208C10.1822 3.69999 10.6187 3.95287 11.007 4.26989C11.0546 4.3081 11.1139 4.32881 11.1749 4.32853C11.2458 4.32893 11.3139 4.30113 11.3642 4.25121L13.2569 2.3585C13.3088 2.30623 13.3368 2.23485 13.3342 2.16122C13.3382 2.08974 13.3132 2.01958 13.2649 1.96666Z" fill="#E4E4EE"/>
</g>
</g>
<defs>
<clipPath id="clip0">
<rect width="16" height="16" fill="white"/>
</clipPath>
</defs>
</svg>
