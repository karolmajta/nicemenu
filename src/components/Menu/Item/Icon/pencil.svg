<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<g opacity="0.5" clip-path="url(#clip0)">
<path d="M1.6 11.04L0 16L4.96 14.4L1.6 11.04Z" fill="#E4E4EE"/>
<path d="M10.53 2.08406L2.72363 9.89037L6.11769 13.2844L13.924 5.47812L10.53 2.08406Z" fill="#E4E4EE"/>
<path d="M15.7602 2.48L13.5202 0.24C13.2002 -0.08 12.7202 -0.08 12.4002 0.24L11.6802 0.96L15.0402 4.32L15.7602 3.6C16.0802 3.28 16.0802 2.8 15.7602 2.48Z" fill="#E4E4EE"/>
</g>
<defs>
<clipPath id="clip0">
<rect width="16" height="16" fill="white"/>
</clipPath>
</defs>
</svg>
