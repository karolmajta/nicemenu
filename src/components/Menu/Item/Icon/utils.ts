import { IconName } from "./types";

import boltUrl from "./bolt.svg";
import dotUrl from "./dot.svg";
import inboxUrl from "./inbox.svg";
import loaderUrl from "./loader.svg";
import pencilUrl from "./pencil.svg";
import rescheduleUrl from "./reschedule.svg";

export const iconMapping: { [k in IconName]: string } = {
  bolt: boltUrl,
  dot: dotUrl,
  inbox: inboxUrl,
  loader: loaderUrl,
  pencil: pencilUrl,
  reschedule: rescheduleUrl,
};
