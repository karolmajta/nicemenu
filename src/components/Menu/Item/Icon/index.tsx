import React from "react";
import cx from "classnames";
import styled from "styled-components";

import { IconProps } from "./types";
import { iconMapping } from "./utils";

export function IconUnstyled({ name, className, classNames }: IconProps) {
  return (
    <div
      data-testid="Laserfocus-Menu-Item-Icon"
      className={cx(
        "Laserfocus-Menu-Item-Icon",
        className,
        ...(classNames || [])
      )}
    >
      <img src={iconMapping[name]} alt={name} />
    </div>
  );
}

const Icon = styled(IconUnstyled)`
  display: flex;
  justify-content: center;
  align-items: center;
  margin-left: 20px;
  margin-right: 10px;

  img {
    display: block;
    width: 16px;
    height: 16px;
  }
`;

export default Icon;
