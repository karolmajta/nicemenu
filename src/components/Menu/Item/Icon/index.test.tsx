import React from "react";
import { render, cleanup } from "@testing-library/react";
import Icon from ".";
import { iconMapping } from "./utils";
import { IconName } from "./types";

test("renders correctly for every allowed icon name", () => {
  Object.values(iconMapping).forEach((iconName) => {
    const { getByTestId } = render(<Icon name={iconName as IconName} />);
    const iconDiv = getByTestId("Laserfocus-Menu-Item-Icon");
    expect(iconDiv).toMatchSnapshot();
    cleanup();
  });
});
