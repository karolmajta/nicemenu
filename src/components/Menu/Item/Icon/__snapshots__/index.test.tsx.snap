// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for every allowed icon name 1`] = `
<div
  class="Laserfocus-Menu-Item-Icon sc-bdnylx uunbO"
  data-testid="Laserfocus-Menu-Item-Icon"
>
  <img
    alt="bolt.svg"
  />
</div>
`;

exports[`renders correctly for every allowed icon name 2`] = `
<div
  class="Laserfocus-Menu-Item-Icon sc-bdnylx uunbO"
  data-testid="Laserfocus-Menu-Item-Icon"
>
  <img
    alt="dot.svg"
  />
</div>
`;

exports[`renders correctly for every allowed icon name 3`] = `
<div
  class="Laserfocus-Menu-Item-Icon sc-bdnylx uunbO"
  data-testid="Laserfocus-Menu-Item-Icon"
>
  <img
    alt="inbox.svg"
  />
</div>
`;

exports[`renders correctly for every allowed icon name 4`] = `
<div
  class="Laserfocus-Menu-Item-Icon sc-bdnylx uunbO"
  data-testid="Laserfocus-Menu-Item-Icon"
>
  <img
    alt="loader.svg"
  />
</div>
`;

exports[`renders correctly for every allowed icon name 5`] = `
<div
  class="Laserfocus-Menu-Item-Icon sc-bdnylx uunbO"
  data-testid="Laserfocus-Menu-Item-Icon"
>
  <img
    alt="pencil.svg"
  />
</div>
`;

exports[`renders correctly for every allowed icon name 6`] = `
<div
  class="Laserfocus-Menu-Item-Icon sc-bdnylx uunbO"
  data-testid="Laserfocus-Menu-Item-Icon"
>
  <img
    alt="reschedule.svg"
  />
</div>
`;
